API documentation for CAN indicator (#422)
cmu-delphi/delphi-epidata: docs/api/covidcast-signals/covid-act-now.md

## Table of contents
{: .no_toc .text-delta}

1. TOC
{:toc}

## Estimation

The quantities received from CAN / CDC are the county-level positivity rate and total tests, 
which are based on the counts of PCR specimens tested.
In particular, they are also already smoothed with a 7-day-average.

For a fixed location $$i$$ and time $$t$$, let $$Y_{it}$$ denote the number of PCR specimens 
tested that have a positive result. Let $$N_{it}$$ denote the total number of PCR specimens tested.
Let $$p_{it}$$ be the PCR-specimen positivity rate as a binomial proportion.

At the county-level, $$p_{it}$$ and $$N_{it}$$ are taken directly from CAN / CDC without modification.
For the sake of aggregating to other geographical levels, we estimate each county's $$Y_{it}$$ as

$$
\hat{Y}_{it} =  N_{it} p_{it}
$$

Let $$G$$ be a set of counties with associated population weights $$w_i$$ for each county $$i \in G$$.
At other geographical levels, the total number of tests $$N_{Gt}$$ 
and our estimate of the positivity rate $$\hat{p}_{Gt}$$ is:

$$
\begin{aligned}
    N_{Gt} &= \sum_{i \in G} w_i N_{it} \\
    \hat{p}_{Gt} &= \frac{ \sum_{i \in G} w_i \hat{Y}_{it} }{ N_{Gt} }
\end{aligned}
$$

> Note that in order to more closely mirror the CAN data source, no pseudo-observations and priors 
> are used in the estimation of these proportions.

Reviewer: ```suggestion
Note that in order to more closely mirror the CAN data source, no pseudo-observations
(such as we would typically apply for a Jeffreys prior)
are used in the estimation of these proportions. Instead of using the Jeffreys prior to
more accurately estimate the standard error when the value is zero, we suppress all
entries with a value of zero.
```

Author: Thanks for the detail! I added on a short bit to clarify that we suppress entries with zero sample size.

### Standard Error

The estimated standard errors are estimated by:

$$
\begin{aligned}
    \hat{se}(p_{it}) &= \sqrt{\frac{p_{it} (1 - p_{it})}{N_{it}}} \\
    \hat{se}(\hat{p}_{Gt}) &= \sqrt{\frac{\hat{p}_{Gt} (1 - \hat{p}_{Gt})}{N_{Gt}}}
\end{aligned}
$$

### Smoothing

No additional smoothing is done to avoid double-smoothing, since the data pulled from CAN / CDC 

> is already smoothed as a 7-day-average.

Reviewer: ```suggestion
is already smoothed with a 7-day-average.
```

## Limitations

Estimates for geographical levels beyond counties may be inaccurate due to how aggregations 
are done on smoothed values instead of the raw values. Ideally we would aggregate raw values 
then smooth, but the raw values are not accessible in this case.

The positivity rate here should not be interpreted as the population positivity rate as 
the testing performed are typically not randomly sampled, especially for early data 
with lower testing volumes.

A few counties, most notably in California, are also not covered by this data source.

> 

Reviewer: A line here repeating that we suppress zeros

## Lag and Backfill

> The lag for these signals vary depending on the reporting patterns of individual counties.

Reviewer: ```suggestion
The lag for these signals varies depending on the reporting patterns of individual counties.
```

Most counties have their latest data report with a lag of 2 days, while others can take 9 days 
or more in the case of California counties.

> These signals are also backfilled as backlogged test results could get assigned to older 7-day timeframes.

Reviewer: Do we know how much backfill there is?

* How many days are submitted in each issue?
* What's the median change from the first day a value is issued to its final update?

Author: Unfortunately I have only pulled a few non-contiguous days of CAN data on my side to have accurate statistics about issue dates.
Will go through them to try get a rough idea, but will also spin up a cron job in the meantime to collect it for a few days.

Author: Ooh ok CAN commits their [parquet files into github](https://github.com/covid-projections/covid-data-public/commits/main/data/can-scrape), so I can go through the commit history to get the statistics.

Author: Regarding how many days are submitted in each issue, we get data from the very start (Mar 2020) every time we pull from CAN, but rely on the archive differ to figure out the differences.
Should that be mentioned in this section too?

Reviewer: No, the description of diff-based issues is covered on the versioning page. This is just to get an idea of the scale of the average revision. For example, both doctor-visits and covid antigen tests update 60 days in the past, every day, while ght only ever updates 1 day, and cases/deaths typically updates 1 day but occasionally updates much larger spans, sometimes 300 days or more. Your paragraph below is sufficient -- nice work!

---
title: COVID Act Now
parent: Data Sources and Signals
grand_parent: COVIDcast Epidata API
---

# COVID Act Now (CAN)
{: .no_toc}

> * **Source name:** `covid-act-now`

Reviewer 2: FYI the data folder on staging has underscores instead of dashes

Reviewer: Fixed in https://github.com/cmu-delphi/covidcast-indicators/pull/916

A few counties, most notably in California, are also not covered by this data source.

Entries with zero total tests performed are also suppressed, even if it was actually the case that 
no tests were performed for the day.

## Lag and Backfill

The lag for these signals varies depending on the reporting patterns of individual counties.
Most counties have their latest data report with a lag of 2 days, while others can take 9 days 
or more in the case of California counties.

These signals are also backfilled as backlogged test results could get assigned to older 7-day timeframes.
Most recent test positivity rates do not change substantially with backfill (having a median delta of close to 0).
However, most recent total tests performed is expected to increase in later data revisions (having a median increase of 7%).

> Values more than 5 days in the past are expected to remain fairly static, with total tests performed 
> having a median increase of 1% of less), as most major revisions have already occurred.

Reviewer: ```suggestion
Values more than 5 days in the past are expected to remain fairly static (with total tests performed 
having a median increase of 1% of less), as most major revisions have already occurred.
```
